openpype/hosts/maya/api/lib.py

```diff
@@ -2622,10 +2624,18 @@ def _get_attrs(node):
             for attr in transform_attrs:
                 cmds.setAttr('{0}.{1}'.format(new_node, attr), lock=False)
 
+            scale_keys = _remove_scale_keys(node)
+
             # Constraints
             delete_bin.extend(cmds.parentConstraint(node, new_node, mo=False))
             delete_bin.extend(cmds.scaleConstraint(node, new_node, mo=False))
 
+            # resetting scale keys on original node
+            for orig_node_attr, frame, _ in scale_keys:
+                log.debug("resetting key::{} at {}".format(orig_node_attr,
+                                                           frame))
+                cmds.setKeyframe(orig_node_attr, time=(frame,))
```

Comment:
Sorry, wait.. what are you doing here? :) You're breaking the scale Constraint logic by keying over the scale constraint and thus making it a blend constraint override which defaults to `1.0`.

What's going on here? What are we trying to solve but are actually breaking with this code?

Reply:
I was trying to workaround this issue https://github.com/ynput/OpenPype/pull/5445#pullrequestreview-1611062862

We were looking into that and Scale keys are breaking `cmds.scaleConstraint`. Why? IDK.
General consensus here was that keys on scale attributes are most likely just side product of keying via `S` and most likely not necessary (fingers crossed).
So the idea is to check if key values are default, eg. just generic `S`, if so, they could be safely removed (and then set back for original camera). If they are different, artist would like to set them, but this is not supported. (And we will need to figure out how to solve it, hopefully it won't happen.)
Open for ideas how to solve that `Could not add constraint or connections` without this unholy mess.

Reply:
I see - this issue is **new** because you're now duplicating the original camera with input connections. As said before - you should avoid that since that's what will get you into this mess. With that disabled the input connections will not be there and thus Scale attribute will be empty. The same issue will occur on translate.

Below [this line](https://github.com/ynput/OpenPype/pull/5445/files#diff-0f94b40dc0b348d7194c61cb01a57668eb3490d22cdbff89d37dd552287fccadR2625) you could just remove any inputs whatsoever to transform attributes - as those should be propagated from the constraints instead to be 'world-space' equivalents.

Using e.g. something like this:
```python
def disconnect_inputs(plug):
    """Disconnect any input sources for the plug."""
    sources = cmds.listConnections(plug, plugs=True, source=True,
                                   destination=True, shapes=True,
                                   skipConversionNodes=False) or []
    if not sources:
        return

    for dest, source in pairwise(sources):
        if cmds.isConnected(source, dest):
            cmds.disconnectAttr(source, dest)
```

And then make it like:
```python
            # Unlock transform attributes so constraint can be created
            for attr in transform_attrs:
                plug = "{}.{}".format(new_node, attr)
                cmds.setAttr(plug, lock=False)
                disconnect_inputs(plug)
```

And voila, you should be able to make the constraints as needed. But as said before - copying input connections can definitely make the behavior much different from what you intend it to be.

I [noted on discord yesterday that there's a flag on `maya.cmds.file` for `exportSelected`](https://discord.com/channels/517362899170230292/563751989075378201/1155954318202118194) which could make this much more trivial, it's [`exportSnapshotCallback`](https://help.autodesk.com/cloudhelp/2020/ENU/Maya-Tech-Docs/CommandsPython/file.html#flagexportSnapshotCallback). Would be quite a refactor but might make it easier in the long run to maintain?



You can just add: `disconnect_inputs(

Reply:
@kalisp not sure why but for me this comment chain doesn't show on the "Changes" page on github - so just commenting here once more to ensure you've seen this comment. :)